RoaringBitmap/src/main/java/org/roaringbitmap/art/LeafNode.java

```diff
@@ -131,4 +131,9 @@ public long getContainerIdx() {
   public byte[] getKeyBytes() {
     return LongUtils.highPart(key);
   }
+
+  @Override
+  public int compareTo(LeafNode other) {
+    return Long.compare(this.key, other.key);
```

Comment:
We need to compare the key by making the low 16 bits as 0.

Reply:
Ignore this comment, the low 16 bits already be 0.

Reply:
Oh, actually I should just remove this again. We don't actually need it anywhere. It's a left-over from me trying to have more general `PeekableIterator` interface, but it got messy instead of helping, so I abandoned it.